lib/routes/v1.js

```diff
@@ -169,6 +183,18 @@ api.declare({
 
   let modDate = new Date();
 
+  // We want to make sure that all AMIs that we are submitting are valid
+  for (let region of input.regions) {
+    let validAmi = await amiExists(this.ec2, region.launchSpec.ImageId, region.region);
```

Comment:
This was written when we were thinking that Promise.all might be related to provisioner freezing and I was avoiding using it.  Yes, we can do it with Promise.all!